Thinktecture.Relay.Server/Repository/UserRepository.cs

```diff
@@ -180,5 +204,63 @@ public bool IsUserNameAvailable(string userName)
 				return user == null;
 			}
 		}
+
+		private DateTime? LockedOutUntil(int? failedAttempts, DateTime? lastFailedAttempt)
+		{
+			if (!failedAttempts.HasValue)
+				return null;
+
+			if (failedAttempts < _configuration.MaxFailedLoginAttempts)
+				return null;
+
+			if (!lastFailedAttempt.HasValue)
+				return null;
+
+			if (lastFailedAttempt + _configuration.FailedLoginLockoutPeriod < DateTime.UtcNow)
```

Comment:
This should be calculated once and used for the check and the return value later.